src/library_manager/lib_manager.c

```diff
@@ -47,6 +51,73 @@ struct lib_manager_dma_ext {
 
 static struct ext_library loader_ext_lib;
 
+#if CONFIG_LIBRARY_AUTH_SUPPORT
+static int lib_manager_auth_init(void)
+{
+	struct ext_library *ext_lib = ext_lib_get();
+	int ret;
+
+	if (auth_api_version().major != AUTH_API_VERSION_MAJOR)
+		return -EINVAL;
+
+	ext_lib->auth_buffer = rballoc_align(0, SOF_MEM_CAPS_RAM,
+					     AUTH_SCRATCH_BUFF_SZ, CONFIG_MM_DRV_PAGE_SIZE);
+	if (!ext_lib->auth_buffer)
+		return -ENOMEM;
+
+	ret = auth_api_init(&ext_lib->auth_ctx, ext_lib->auth_buffer,
+			    AUTH_SCRATCH_BUFF_SZ, IMG_TYPE_LIB);
+	if (ret != 0) {
+		tr_err(&lib_manager_tr, "lib_manager_auth_init() failed with error: %d", ret);
+		ret = -EACCES;
+	}
+
+	return ret;
+}
+
+static void lib_manager_auth_deinit(void)
+{
+	struct ext_library *ext_lib = ext_lib_get();
+
+	if (ext_lib->auth_buffer)
+		memset(ext_lib->auth_buffer, 0, AUTH_SCRATCH_BUFF_SZ);
+
+	rfree(ext_lib->auth_buffer);
+	ext_lib->auth_buffer = NULL;
+	memset(&ext_lib->auth_ctx, 0, sizeof(struct auth_api_ctx));
+}
+
+static int lib_manager_auth_proc(const void *buffer_data,
+				 size_t buffer_size, enum auth_phase phase)
+{
+	struct ext_library *ext_lib = ext_lib_get();
+	int ret;
+
+	ret = auth_api_init_auth_proc(&ext_lib->auth_ctx, buffer_data, buffer_size, phase);
+
+	if (ret != 0) {
+		tr_err(&lib_manager_tr, "lib_manager_auth_proc() failed with error: %d", ret);
+		return -EACCES;
+	}
+
+	/* The auth_api_busy() will timeouts internally in case of failure */
+	while (auth_api_busy(&ext_lib->auth_ctx))
+		;
```

Comment:
why not move 105 to 104 with:
while (auth_api_busy(&ext_lib->auth_ctx));

Reply:
The checkpatch is complaining when the semicolon is in the same line. I found few places in SOF where the same trick was done to satisfy checkpatch, I guess.